### ISSUE TITLE
Constructor can start selecting the bucketlist size at 2 instead of 1

### ISSUE THREAD

**Issue by Illation** [rigtorp/HashMap#3]
Assuming someone constructed a hash map with a capcity of 0 or 1 elements, and proceeded to add an element, the map would immediately rehash. Since we do not start with a completely empty bucket list anyway, we may as well start at 2. This would also save an iteration of the while loop.

**Comment by rigtorp** [comment]
Right, it would make sense to either start with not allocating the item storage and/or to start with a minimum of 8 or something reasonable.
